provider/provider.go

```diff
@@ -62,6 +62,7 @@ func DatabricksProvider() *schema.Provider {
 			"databricks_instance_pool":           pools.DataSourceInstancePool(),
 			"databricks_jobs":                    jobs.DataSourceJobs(),
 			"databricks_job":                     jobs.DataSourceJob(),
+			"databricks_metastores":              catalog.DataSourceMetastores(),
```

Comment:
should we have `databricks_metastores` or `databricks_metastore`?

Reply:
most likely metastore`s` since there can be multiple of them but highlighting for visibility

Reply:
It should be `metastores` because it returns a list. A standard pattern usually is to have two data sources - one returning a list, and another searching for a specific resource and throwing an error if it's not found

Reply:
I will add that in another PR as it will require docs and tests and we can merge this before that.